internal/provider/kubernetes/status.go

```diff
@@ -564,34 +592,3 @@ func (r *gatewayAPIReconciler) updateStatusForGateway(ctx context.Context, gtw *
 		}),
 	})
 }
-
```

Comment:
what about https://github.com/envoyproxy/gateway/blob/71c0b5408f7f80ef8b9d1e0bcf511891839edecd/internal/provider/kubernetes/status.go#L524

Reply:
I suspect the call to `updateStatusForGateway` from inside the predicate blocked the reconciler.

Reply:
Aren't we deleting update status for gateway ?

Reply:
No, the update is stored into the r.resources.GatewayStatuses instead.  The update event will be handled in the status updater which subscripts the GatewayStatuses.